MsApplicationPkg/SecureBootRecovery/SecureBootRecovery.c

```diff
@@ -11,15 +11,64 @@
 #include <Guid/ImageAuthentication.h>
 #include <Guid/VariableFormat.h>
 #include <Library/BaseLib.h>
+#include <Library/BaseMemoryLib.h>
 
 #include "RecoveryPayload.h"
 
+//
 // 10 seconds in microseconds
+//
 #define STALL_10_SECONDS  10000000
 
+//
+// The size of an EFI_STATUS in bytes
+//
 #define STATUS_SIZE         (sizeof(EFI_STATUS) * 2)
 #define STATUS_STRING_SIZE  (STATUS_SIZE + sizeof(L"\0"))
 
+//
+// The certificate we are looking for
+//
+#define CERT_ORGANIZATION         "Microsoft Corporation"
+#define CERT_ORGANIZATION_OFFSET  0xF6
+#define CERT_COMMON_NAME          "Windows UEFI CA 2023"
+#define CERT_COMMON_NAME_OFFSET   0x116
+#define CERT_SIGNATURE            { 0x9f, 0xc9, 0xb6, 0xff, 0x6e, 0xe1, 0x9c, 0x3b, 0x55, 0xf6, 0xfe, 0x8b, 0x39, 0xdd, 0x61, 0x04 }
+#define CERT_SIGNATURE_OFFSET     0x3AE
+
+//
+// The system table pointers
+//
+EFI_SYSTEM_TABLE      *mST = NULL;
+EFI_RUNTIME_SERVICES  *mRT = NULL;
+EFI_BOOT_SERVICES     *mBS = NULL;
+
+typedef struct {
+  UINT8    *Organization;      // The organization name in the certificate

```

Comment:
I don't know the details of this string's contents, but was there a particular reason for using `UINT8` vs `CHAR8`?

Reply:
It looks like it is expected to be an ASCII string like `"Microsoft Corporation"`? In that case, I'd recommend `CHAR8`.